### DISCUSSION TITLE
how to use a custom font family with this library?

### DISCUSSION THREAD

**Discussion by Yasir5247** [wix/react-native-ui-lib#2081] [Q&A] [answered]
I can see that i can control fontSize, fontWeight etc. for instance "text90", "text70" is for different sizes. also i can make it bold and large like this "text100BL". but where is the font family defined?. i assume this is the system default font. I would like to change it to a font family of my choice where i can control the weight, size like this using these presets. Is it possible? if so how?

```
 return (
    <View marginB-10>
      <View>
        {!!heading && (
          <Text text90 grey40>
            {heading}
          </Text>
        )}
      </View>
      <View row>
        <View>
          <Text text70 black>
            {replaceIdNumber(idNumber)}
          </Text>
        </View>
      </View>
    </View>
  );
```

**Comment by ethanshar** [comment, marked as answer]
You can set your own typography presets or override the existing ones and pass whatever font prop you want: `fontSize`, `fontFamily`, `fontWeight`, `lineHeight`, `letterSpacing`, etc..

You can use `Typography.loadTypographies({myFont: {...}, text70: {...}});`

It's important to notice that Android and iOS works a little different when it comes to font. 
Sometimes the weight of a font is determined by the `fontFamily` and sometimes you need to pass the `fontWeight` 

You can see for example how we set our internal typographies here
https://github.com/wix/react-native-ui-lib/blob/cf29b80a6e8d216794db971c8419228d4c4db365/src/style/typographyPresets.ts#L19
